client/components/data/query-wordads-earnings/index.jsx

```diff
@@ -0,0 +1,40 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import PropTypes from 'prop-types';
+import { Component } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { requestWordadsEarnings } from 'state/wordads/earnings/actions';
+
+class QueryWordadsEarnings extends Component {
+	static propTypes = {
+		requestWordadsEarnings: PropTypes.func,
+		siteId: PropTypes.number,
+	};
+
+	static defaultProps = {
+		requestWordadsEarnings: () => {},
+	};
+
+	componentDidMount() {
+		this.props.requestWordadsEarnings( this.props.siteId );
+	}
+
+	componentDidUpdate( prevProps ) {
+		if ( this.props.siteId !== prevProps.siteId ) {
+			this.props.requestWordadsEarnings( this.props.siteId );
+		}
+	}
+
+	render() {
+		return null;
+	}
+}
+
+export default QueryWordadsEarnings;
```

Comment:
There should be a `connect` call here that provides the `requestWordadsEarnings` prop. If it's missing, the prop is undefined.